File: android-core/src/androidTest/java/com/mparticle/internal/database/services/BaseMPServiceTest.java
package com.mparticle.internal.database.services;

import android.database.sqlite.SQLiteOpenHelper;
import android.location.Location;
import androidx.test.platform.app.InstrumentationRegistry;

import com.mparticle.internal.InternalSession;
import com.mparticle.internal.database.BaseDatabase;
import com.mparticle.internal.database.MPDatabaseImpl;
import com.mparticle.internal.database.tables.MParticleDatabaseHelper;
import com.mparticle.internal.messages.BaseMPMessage;
import com.mparticle.testutils.BaseCleanInstallEachTest;

import org.json.JSONException;
import org.junit.Before;

import java.util.UUID;

abstract public class BaseMPServiceTest extends BaseCleanInstallEachTest {
    protected static MPDatabaseImpl database;

    @Before
    public final void beforeBaseMPService() throws Exception {
        SQLiteOpenHelper openHelper = new BaseDatabase(new MParticleDatabaseHelper(mContext), MParticleDatabaseHelper.DB_NAME);
        database = new MPDatabaseImpl(openHelper.getWritableDatabase());
    }

    BaseMPMessage getMpMessage() throws JSONException {
        return getMpMessage(UUID.randomUUID().toString());
    }

    BaseMPMessage getMpMessage(String sessionId) throws JSONException {
        return getMpMessage(sessionId, mRandomUtils.randomLong(Long.MIN_VALUE, Long.MAX_VALUE));
    }

    BaseMPMessage getMpMessage(String sessionId, long mpid) throws JSONException {
        InternalSession session = new InternalSession();
        session.mSessionID = sessionId;
        return new BaseMPMessage.Builder(mRandomUtils.getAlphaNumericString(mRandomUtils.randomInt(20, 48))).build(session, new Location(mRandomUtils.getAlphaNumericString(mRandomUtils.randomInt(1, 55))), mpid);
    }
}

File: android-core/src/androidTest/java/com/mparticle/internal/database/services/BaseMPServiceTest.kt
package com.mparticle.internal.database.services

import android.database.sqlite.SQLiteOpenHelper
import android.location.Location
import com.mparticle.internal.InternalSession
import com.mparticle.internal.database.MPDatabaseImpl
import com.mparticle.internal.database.TestSQLiteOpenHelper
import com.mparticle.internal.database.tables.MParticleDatabaseHelper
import com.mparticle.internal.messages.BaseMPMessage
import com.mparticle.testing.BaseTest
import com.mparticle.testing.RandomUtils
import com.mparticle.testing.context
import org.json.JSONException
import org.junit.Before
import java.util.UUID

abstract class BaseMPServiceTest : BaseTest() {
    @Before
    @Throws(Exception::class)
    fun beforeBaseMPService() {
        val openHelper: SQLiteOpenHelper = TestSQLiteOpenHelper(
            MParticleDatabaseHelper(context),
            MParticleDatabaseHelper.getDbName()
        )
        database = MPDatabaseImpl(openHelper.writableDatabase)
    }

    @get:Throws(JSONException::class)
    val mpMessage: BaseMPMessage
        get() = getMpMessage(UUID.randomUUID().toString())

    @Throws(JSONException::class)
    fun getMpMessage(sessionId: String): BaseMPMessage {
        return getMpMessage(sessionId, RandomUtils.randomLong(Long.MIN_VALUE, Long.MAX_VALUE))
    }

    @Throws(JSONException::class)
    fun getMpMessage(sessionId: String, mpid: Long): BaseMPMessage {
        val session = InternalSession()
        session.mSessionID = sessionId
        return BaseMPMessage.Builder(
            RandomUtils.getAlphaNumericString(
                RandomUtils.randomInt(
                    20,
                    48
                )
            )
        ).build(
            session,
            Location(RandomUtils.getAlphaNumericString(RandomUtils.randomInt(1, 55))),
            mpid
        )
    }

    companion object {
        @JvmStatic
        protected var database: MPDatabaseImpl? = null
    }
}
